client/db/test/types_test.go

```diff
@@ -40,11 +40,14 @@ func TestMatchProof(t *testing.T) {
 	nTimes(spins, func(i int) {
 		proof := proofs[i]
 		proofB := proof.Encode()
-		reProof, err := db.DecodeMatchProof(proofB)
+		reProof, ver, err := db.DecodeMatchProof(proofB)
 		if err != nil {
 			t.Fatalf("match decode error: %v", err)
 		}
 		MustCompareMatchProof(t, proof, reProof)
+		if ver != db.MatchProofVer {
+			t.Errorf("wanted match proof ver %d, got %d", db.MatchProofVer, ver)
+		}
```

Comment:
I'm actually not getting why either the v1 or v3 upgrades that just call `reloadMatchProofs` are needed at all given the MatchProofs will always load the same regardless of the serialization version in the stored data.